mlflow/store/tracking/file_store.py

```diff
@@ -1319,3 +1326,363 @@ def _read_helper(root, file_name, attempts_remaining=2):
                 return _read_helper(root, file_name, attempts_remaining - 1)
 
         return _read_helper(root, file_name, attempts_remaining=retries)
+
+    def start_trace(
+        self,
+        experiment_id: str,
+        timestamp_ms: int,
+        request_metadata: Dict[str, str],
+        tags: Dict[str, str],
+    ) -> TraceInfo:
+        """
+        Start an initial TraceInfo object in the backend store.
+
+        Args:
+            experiment_id: String id of the experiment for this run.
+            timestamp_ms: Start time of the trace, in milliseconds since the UNIX epoch.
+            request_metadata: Metadata of the trace.
+            tags: Tags of the trace.
+
+        Returns:
+            The created TraceInfo object.
+        """
+        request_id = request_metadata.get(SpanAttributeKey.REQUEST_ID)
+        if request_id is None:
+            raise MlflowException("Request ID is required to start a trace.", INTERNAL_ERROR)
+        _validate_experiment_id(experiment_id)
+        experiment_dir = self._get_experiment_path(
+            experiment_id, view_type=ViewType.ACTIVE_ONLY, assert_exists=True
+        )
+        mkdir(experiment_dir, FileStore.TRACES_FOLDER_NAME)
+        traces_dir = os.path.join(experiment_dir, FileStore.TRACES_FOLDER_NAME)
+        mkdir(traces_dir, request_id)
```

Comment:
WDYT about using os.path.mkdirs(<fully qualified trace_dir path>, exist_ok=True)?

Reply:
mkdir internally calls os.path.mkdirs and catch EEXIST error without raising, I'll update it to use exist_ok=True instead!